/* Copyright 2022 The Waymo Open Dataset Authors. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto2";

package waymo.open_dataset;

import "waymo_open_dataset/protos/segmentation_metrics.proto";

// If your inference results are too large to fit in one proto, you can shard
// them to multiple files by sharding the inference_results field.
// Next ID: 11.
message SemanticSegmentationSubmission {
  enum SensorType {
    INVALID = 0;
    LIDAR_ALL = 1;
    LIDAR_TOP = 2;
    CAMERA_ALL = 3;
    CAMERA_LIDAR_TOP = 4;
    CAMERA_LIDAR_ALL = 5;
  }

  // This must be set as the full email used to register at waymo.com/open.
  optional string account_name = 1;
  // This name needs to be short, descriptive and unique. Only the latest result
  // of the method from a user will show up on the leaderboard.
  optional string unique_method_name = 2;
  repeated string authors = 3;
  optional string affiliation = 4;
  optional string description = 5;

  // Link to paper or other link that describes the method.
  optional string method_link = 6;

  optional SensorType sensor_type = 7;
  // Number of frames used.
  optional int32 number_past_frames_exclude_current = 8;
  optional int32 number_future_frames_exclude_current = 9;

  // Inference results.
  optional SegmentationFrameList inference_results = 10;
}
